### DISCUSSION TITLE
keygen self hosted on AWS

### DISCUSSION THREAD

**Discussion by bodev1406** [keygen-sh/community#89] [Q&A]
Hi guys,

we have to host keygen by ourselve because of some data privacy stuff. We want to host it on AWS. For the Docker container we want to use AWS Fargate. For Redis, AWS Elasticache. For the Postgres DB, AWS RDS.
To setup the keygen container we have to run a setup command. 
Does anyone know how to do it on AWS and Fargate? 

Besides that, does anyone has some experience hosting keygen on AWS?

**Comment by ezekg** [comment]
You could technically run the `setup` command locally to provision the database and to generate the appropriate environment variables, and then configure those environment variables in AWS Fargate. Just make sure you're locally connected to the RDS database.

After that, Keygen can be deployed [like any other Rails app on Fargate](https://www.honeybadger.io/blog/rails-docker-aws-fargate/).

**Comment by bodev1406** [comment]
Ok, thanks for you response.

Another question, is it possible to run to instances of keygen? I mean to use AWS autoscaling to run more instances.

**Reply by ezekg** [reply]
Of course. As long as your Postgres and Redis databases can scale connections, you can scale as much as you want.
